<a href="https://colab.research.google.com/github/Datebi/CursoCoderhouse/blob/main/Entrega1_Baigorria.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>



---


>>>>> #      Primer entrega del proyecto final 

>> # "Estudio sobre riesgos crediticios en institución bancaria"

>>Profesor: David Romero Acosta 

>>Tutor: Enrique Corimayo

>>Alumno: Esteban Baigorria

>*- Data Science - Coderhouse -*

---



# >Objetivos Generales:

*   Entender el problema de negocio e identificar los elementos a ser considerados para el planteamiento de un Modelo de Data Science.
*   Describir los datos de negocio y las relaciones entre datos mediante el Análisis Exploratorio de Datos.
*   Construir una presentación ejecutiva para la alta gerencia mostrando los resultados obtenidos.

---



# >Objetivos Específicos:

*   Desarrollar las instancias de Data Acquisition y Data Wrangling en tu trabajo final.

*   Lograr una articulación en equipo y una división de tareas adecuadas a los objetivos.

*   Realizar Filtrado.

*   Describir qué significa cada variable, cómo se comporta.
Especificar las distribuciones y relaciones (géneros, sexo, edad, IVA, tipo de empresa).


---



# >Se debe entregar:

*   Presentación de la empresa, organización o problema específico.

*   Preguntas y objetivos de la investigación.

*   Conformación del equipo de trabajo.

*   Indicación de la fuente del dataset y los criterios de selección (Data Acquisition).

*   Generación del primer Data Wrangling y EDA, apuntado a sus datos (insights) univariado, bivariado y multivariado.

*   Análisis de componentes principales.

*   Contar la historia de sus datos

*   Filtros aplicados a los datos. Distribución. Dataset final para analizar.

*   Palttear objetivos u objetivo para esos datos.


---




---



---



---



> Una institución bancaria que ofrece creditos al publico se presenta en nuestras oficinas y solicita una evaluacion en base a su base de datos para conocer si es posible clasificar a un cliente como riesgoso o no riesgo en productos bancarios especificos. A continuacion se presenta el click al dataset

>>"Credit Risk Classification Dataset":
>>https://www.kaggle.com/praveengovi/credit-risk-classification-dataset 



> El dataset se compone de las transacciones y datos demograficos. Ademas clasifica en cliente riesgoso y no riesgoso para productos especificos del banco.




>En referencia a los dos archivos csv que componen el dataset:

>>payment_data.csv:

>>>**paymentdata.csv:** Contiene al historial de pago de la tarjeta del cliente

>>>**id:** Numero entero identificador del cliente

>>>**OVDt1:** Cantidad de veces con fecha de pagos vencidos del tipo 1.

>>>**OVDt2:** Cantidad de veces con fecha de pagos vencidos del tipo 2.

>>>**OVDt3:** Cantidad de veces con fecha de pagos vencidos del tipo 3.

>>>**OVDsum:** Cantidad total de de dias que concurrieron con el vencimiento del 
pago vencido.

>>>**paynormal:** Cantidad de dias con fecha de pago normal.

>>>**prodcode:** Codigo de producto de credito

>>>**prodlimit:** Limite de credito del producto.

>>>**updatedate:** Fecha de actualizacion de la cuenta.

>>>**newbalance:** Balance actual del producto.

>>>**highestbalance:** Balance mas alto historico.

>>>**reportdate:** Fecha de pago mas reciente.

>>customer_data.csv:

>>>**customer_data.csv:** Contiene datos demograficos y categoricos sobre los clientes totalmente encriptados para evitar el sesgo, se utilizara la denominacion 'fea' para referirnos a ellas (feature encrypted attribute).

>>>Las variables categoricas son los campos: **fea1, fea3, fea5, fea6, fea7, fea9. Las demograficas: fea2, fea10 y fea11**.

>>>**Label:** 1 si el cliente es de alto riesgo crediticio, 0 si corresponde a un cliente de bajo riesgo.



>*Fuente del Dataset: Google Search*

>*Criterio de selección: Download*



---



---



---



El objetivo del proyecta será poder contestar:

    1.Cuales son los factores contribuyendo a un cliente con alto riesgo crediticio?
    2.Cuales son los atributos de un cliente de bajo riesgo?
    3.Con cuanta precision el modelo podra predecir si un cliente sera de riesgo alto o bajo. 




---



---



---





In [ ]:
## La primera parte del proyecto correspondera a la importación de los sets de datos, data cleaning, al data wrangling y al EDA.
## Se procedera con la siguiente estructura de carpetas

## Directorio_de_trabajo
## |
## |
## |----> Data
## |
## |
## |----> Entrenamiento
## |
## |
## |----> Test
## |
## |
## |----> Validación
## |
## |
## |----> Visuales 

In [1]:
## Importamos las librerias necesarias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import shutil
import os
import zipfile
import matplotlib.pyplot as plt
from matplotlib.ticker import FormatStrFormatter


>Descarga de datos desde la fuente.

>Los set de datos utilizados en este proyecto seran descargados utilizando la API de Kaggle desde la siguiente direccion:

>>https://www.kaggle.com/praveengovi/credit-risk-classification-dataset

In [21]:
# Descomentar las lineas con una # para bajar el dataset directamente desde Kaggle
## Para instalar el paquete de Kaggle en caso de no tenerlo previamente:
# !pip install kaggle
## Descarga del dataset:
# !echo '{"username":"USERNAME","key":"KEY"}' > /root/.kaggle/kaggle.json #Es posible que deba correr esta linea 2 veces
# !kaggle datasets download -d praveengovi/credit-risk-classification-dataset


  0% 0.00/148k [00:00<?, ?B/s]
100% 148k/148k [00:00<00:00, 1.33MB/s]


In [7]:
## Imprimimos el directorio actual para ubicar nuestros ficheros

print(os.getcwd())

/content


In [8]:
## Asignamos el directorio actual a una variable para crear las carpetas
ruta = os.getcwd()
print(ruta)



/content


In [9]:
## Creamos la carpeta de data
os.mkdir(os.path.join(ruta, 'data'))

In [10]:
## Creamos la carpeta de entrenamiento
os.mkdir(os.path.join(ruta, 'entrenamiento'))

In [11]:
## Creamos la carpeta de test 
os.mkdir(os.path.join(ruta, 'test'))

In [12]:
## Creamos la carpeta de validaciones
os.mkdir(os.path.join(ruta, 'validacion'))

In [13]:
## Creamos la carpeta de visuales 
os.mkdir(os.path.join(ruta, 'visuales'))

In [22]:
## Movemos el archivo zip que contiene los csv a nuestra carpeta de data
source = os.path.join(ruta, 'credit-risk-classification-dataset.zip')
destination = os.path.join(ruta, 'data', 'CreditosBancariosDS.zip')
shutil.move(source, destination)



'/content/data/CreditosBancariosDS.zip'

In [23]:
## Chequeamos el contenido de nuestra carpeta data
print(os.listdir(os.path.join(ruta, 'data')))



['CreditosBancariosDS.zip', '.ipynb_checkpoints']


In [24]:
## Descomprimimos al archivo que contiene a los datasets
with zipfile.ZipFile(destination, 'r') as zip_ref:
    zip_ref.extractall(os.path.join(ruta, 'data'))



In [25]:
## Chequeamos el contenido de nuestra carpeta data para verificar que se hayan descomprimido correctamente
print(os.listdir(os.path.join(ruta, 'data')))


['CreditosBancariosDS.zip', '.ipynb_checkpoints', 'customer_data.csv', 'payment_data.csv']


In [26]:
## Comenzamos con la limpieza de datos de los archivos CSV, siendo el primer archivo a limpiar el de customer_data:
## Generamos el dataframe con nuestros datos 
df_clientes = pd.read_csv('data/customer_data.csv')

In [27]:
## Chqueamos el shape de nuestros df
shapecliente = df_clientes.shape
print("El shape en (filas,columnas) de df de clientes es",shapecliente)

El shape en (filas,columnas) de df de clientes es (1125, 13)


In [28]:
## Tipos de datos
df_clientes.dtypes

label       int64
id          int64
fea_1       int64
fea_2     float64
fea_3       int64
fea_4     float64
fea_5       int64
fea_6       int64
fea_7       int64
fea_8       int64
fea_9       int64
fea_10      int64
fea_11    float64
dtype: object

In [29]:
## Mostramos las primeros 5 filas para verificar su correcta carga y tener un primer vistazo a los datos:
df_clientes.head(5)

label        id  fea_1   fea_2  ...  fea_8  fea_9  fea_10      fea_11
0      1  54982665      5  1245.5  ...    109      5  151300  244.948974
1      0  59004779      4  1277.0  ...    100      3  341759  207.173840
2      0  58990862      7  1298.0  ...    101      5   72001    1.000000
3      1  58995168      7  1335.5  ...    110      3   60084    1.000000
4      0  54987320      7     NaN  ...    108      4  450081  197.403141

[5 rows x 13 columns]

In [30]:
## Mostramos las ultimas 5 filas para verificar su correcta carga y tener un segundo vistazo a los datos::
df_clientes.tail(5)         

label        id  fea_1   fea_2  ...  fea_8  fea_9  fea_10      fea_11
1120      0  58988196      5  1289.0  ...    112      3  350702  200.000000
1121      0  58987926      5     NaN  ...    108      4  450000  169.000000
1122      0  58995381      7  1220.0  ...     90      5   71002    1.000000
1123      0  58998054      4  1250.0  ...     90      5   72000    1.000000
1124      0  54989781      4  1415.0  ...    113      4  151300  273.861279

[5 rows x 13 columns]

In [31]:
## Chequeamos el resumen estadistico del df
df_clientes.describe(include = 'all')

label            id  ...         fea_10       fea_11
count  1125.000000  1.125000e+03  ...    1125.000000  1125.000000
mean      0.200000  5.783677e+07  ...  164618.495111   134.999004
std       0.400178  1.817150e+06  ...  152520.488281   112.616798
min       0.000000  5.498235e+07  ...   60000.000000     1.000000
25%       0.000000  5.499050e+07  ...   60044.000000     1.000000
50%       0.000000  5.898975e+07  ...   72000.000000   173.205081
75%       0.000000  5.899799e+07  ...  151307.000000   202.484567
max       1.000000  5.900624e+07  ...  650070.000000   707.106781

[8 rows x 13 columns]

In [32]:
## Realizamos un pivot de la funcion describe para visualizar mejor nuestros datos!
df_clientes.describe(include = 'all').T

count          mean  ...           75%           max
label   1125.0  2.000000e-01  ...  0.000000e+00  1.000000e+00
id      1125.0  5.783677e+07  ...  5.899799e+07  5.900624e+07
fea_1   1125.0  5.482667e+00  ...  7.000000e+00  7.000000e+00
fea_2    976.0  1.283911e+03  ...  1.314500e+03  1.481000e+03
fea_3   1125.0  2.333333e+00  ...  3.000000e+00  3.000000e+00
fea_4   1125.0  1.208836e+05  ...  1.390000e+05  1.200000e+06
fea_5   1125.0  1.928889e+00  ...  2.000000e+00  2.000000e+00
fea_6   1125.0  1.087200e+01  ...  1.100000e+01  1.600000e+01
fea_7   1125.0  4.832889e+00  ...  5.000000e+00  1.000000e+01
fea_8   1125.0  1.008027e+02  ...  1.110000e+02  1.150000e+02
fea_9   1125.0  4.195556e+00  ...  5.000000e+00  5.000000e+00
fea_10  1125.0  1.646185e+05  ...  1.513070e+05  6.500700e+05
fea_11  1125.0  1.349990e+02  ...  2.024846e+02  7.071068e+02

[13 rows x 8 columns]

In [33]:
## Chequeamos en la informacion de clientes si los id son unicos para corroborar que pueda ser convertido en un indice.
df_clientes['id'].nunique()

1125

> De esta manera se confirma que los valores del campo 'id' son únicos

In [34]:
## Buscamos valores duplicados
duplicate = df_clientes[df_clientes.duplicated()]
duplicate.head(5)



Empty DataFrame
Columns: [label, id, fea_1, fea_2, fea_3, fea_4, fea_5, fea_6, fea_7, fea_8, fea_9, fea_10, fea_11]
Index: []

In [35]:
## Buscamos si hay valores nulos
df_clientes.isnull().values.any()



True

Al ser 'True' hay que tomar una decision respecto al tratamiento de los valores nulos.

In [ ]:
##Buscamos en que columna hay valores nulos

#df_clientes['id'].isnull().values.any() # Arrojo como resultado False
#df_clientes['label'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_1'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_2'].isnull().values.any() # Arrojo como resultado Tue
#df_clientes['fea_3'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_4'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_5'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_6'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_7'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_8'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_9'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_10'].isnull().values.any() # Arrojo como resultado False
#df_clientes['fea_11'].isnull().values.any() # Arrojo como resultado False



False

> El campo 'fea_2' es el único que tiene valores nulos, además, se observa en la descripción que a diferencia de los otros campos que tienen 1125, contiene 976 valores.

>> El campo 'fea_2' corresponde a un valor flotante.

>> Al ser un valor demografico encriptado no se tiene mayor certeza sobre el mismo mas que podria ser una de las siguientes caracteristicas: Edad,sexo, lugar de residencia, ocupación,situación familiar o ingresos.

>> La mayoria de las caracteristicas quedan descartadas, siendo las unicas dos posibilidades: Lugar de residencia(coordenadas) o ingresos, en cuyo caso ambos valores medios pueden ser buena referencia para reemplazar los valores faltantes en el dataset.

In [36]:
df_clientes['fea_2'].describe().T

count     976.000000
mean     1283.911373
std        51.764022
min      1116.500000
25%      1244.000000
50%      1281.500000
75%      1314.500000
max      1481.000000
Name: fea_2, dtype: float64

La desviación standard es de 51 unidades lo cual representa un 4% de variación respecto al promedio.

CV < 5%, a priori se concluye una variable del tipo homogenea.

Se calcula que la diferencia entre el valor maximo y el minimo en la muestra es de aproximadamente 15%, con lo cual no hay grandes outliers.

Se puede concluir aunque hace falta algo mas de profundización que es una variable que puede ser correctamente reemplazada por su promedio.

In [37]:
## Buscamos en el describe realizado en un paso previo el promedio de 
## la columna antes de realizar el reemplazo para chequear que de igual despues de realizarlo.
## Promedio fea_2 resultado:1283.911

promedio_columna = df_clientes['fea_2'].mean()
df_clientes = df_clientes.fillna(promedio_columna)


In [38]:
## Volvemos a consultar los valores nulos de la columna
df_clientes['fea_2'].isnull().values.any()


False

In [39]:
## Volvemos a describir al df para ver que se muestre bien el conteo de filas
df_clientes.describe(include = 'all').T

count          mean  ...           75%           max
label   1125.0  2.000000e-01  ...  0.000000e+00  1.000000e+00
id      1125.0  5.783677e+07  ...  5.899799e+07  5.900624e+07
fea_1   1125.0  5.482667e+00  ...  7.000000e+00  7.000000e+00
fea_2   1125.0  1.283911e+03  ...  1.310000e+03  1.481000e+03
fea_3   1125.0  2.333333e+00  ...  3.000000e+00  3.000000e+00
fea_4   1125.0  1.208836e+05  ...  1.390000e+05  1.200000e+06
fea_5   1125.0  1.928889e+00  ...  2.000000e+00  2.000000e+00
fea_6   1125.0  1.087200e+01  ...  1.100000e+01  1.600000e+01
fea_7   1125.0  4.832889e+00  ...  5.000000e+00  1.000000e+01
fea_8   1125.0  1.008027e+02  ...  1.110000e+02  1.150000e+02
fea_9   1125.0  4.195556e+00  ...  5.000000e+00  5.000000e+00
fea_10  1125.0  1.646185e+05  ...  1.513070e+05  6.500700e+05
fea_11  1125.0  1.349990e+02  ...  2.024846e+02  7.071068e+02

[13 rows x 8 columns]

In [40]:
## Observemos una fila del dataset en detalle
pd.set_option('display.max_colwidth', -1)
df_clientes.head(1)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.
  


label        id  fea_1   fea_2  ...  fea_8  fea_9  fea_10      fea_11
0  1      54982665  5      1245.5  ...  109    5      151300  244.948974

[1 rows x 13 columns]

In [43]:
## Para facilitar la lectura, vamos a quedarnos con los 4 digitos significativos en fea_11
df_clientes['fea_11'] = df_clientes['fea_11'].round(decimals = 4)

In [44]:
df_clientes

label        id  fea_1        fea_2  ...  fea_8  fea_9  fea_10    fea_11
0     1      54982665  5      1245.500000  ...  109    5      151300  244.9490
1     0      59004779  4      1277.000000  ...  100    3      341759  207.1738
2     0      58990862  7      1298.000000  ...  101    5      72001   1.0000  
3     1      58995168  7      1335.500000  ...  110    3      60084   1.0000  
4     0      54987320  7      1283.911373  ...  108    4      450081  197.4031
...  ..           ... ..              ...  ...  ...   ..         ...       ...
1120  0      58988196  5      1289.000000  ...  112    3      350702  200.0000
1121  0      58987926  5      1283.911373  ...  108    4      450000  169.0000
1122  0      58995381  7      1220.000000  ...  90     5      71002   1.0000  
1123  0      58998054  4      1250.000000  ...  90     5      72000   1.0000  
1124  0      54989781  4      1415.000000  ...  113    4      151300  273.8613

[1125 rows x 13 columns]

In [ ]:
## De esta forma concluye el Data Wrangling del primer archivo csv y comenzamos con el segundo:

In [45]:
## Generamos el dataframe con nuestros datos 
df_historial = pd.read_csv('data/payment_data.csv')



In [46]:
## Chequeamos el shape de nuestros df
shapecliente = df_historial.shape
print("El shape en (filas,columnas) de df de clientes es",shapecliente)

El shape en (filas,columnas) de df de clientes es (8250, 12)


In [47]:
## Tipos de datos
df_historial.dtypes

id                 int64  
OVD_t1             int64  
OVD_t2             int64  
OVD_t3             int64  
OVD_sum            int64  
pay_normal         int64  
prod_code          int64  
prod_limit         float64
update_date        object 
new_balance        float64
highest_balance    float64
report_date        object 
dtype: object

In [48]:
## Mostramos las primeros 5 filas para verificar su correcta carga y tener un primer vistazo a los datos:
df_historial.head(5)

id  OVD_t1  OVD_t2  ...  new_balance  highest_balance  report_date
0  58987402  0       0       ...  0.0         NaN               NaN        
1  58995151  0       0       ...  588720.0     491100.0         NaN        
2  58997200  0       0       ...  840000.0     700500.0         22/04/2016 
3  54988608  0       0       ...  8425.2       7520.0           25/04/2016 
4  54987763  0       0       ...  15147.6     NaN               26/04/2016 

[5 rows x 12 columns]

In [49]:
## Mostramos las ultimas 5 filas para verificar su correcta carga y tener un segundo vistazo a los datos::
df_historial.tail(5)    

id  OVD_t1  OVD_t2  ...  new_balance  highest_balance  report_date
8245  58995478  0       0       ...  0.0         NaN               NaN        
8246  54992408  0       0       ...  0.0         NaN               NaN        
8247  54988209  0       0       ...  20654.4      33315.0          NaN        
8248  54992408  0       0       ...  0.0         NaN               NaN        
8249  54989207  0       0       ...  240000.0     200500.0         NaN        

[5 rows x 12 columns]

In [50]:
## Chequeamos el resumen estadistico del df
df_historial.describe(include = 'all')

id       OVD_t1  ...  highest_balance  report_date
count   8.250000e+03  8250.000000  ...  7.841000e+03     7136       
unique NaN           NaN           ... NaN               1862       
top    NaN           NaN           ... NaN               21/09/2015 
freq   NaN           NaN           ... NaN               103        
mean    5.782173e+07  0.249091     ...  2.192027e+05     NaN        
std     1.822724e+06  1.250197     ...  2.814536e+06     NaN        
min     5.498235e+07  0.000000     ...  5.010000e+02     NaN        
25%     5.499050e+07  0.000000     ...  2.345300e+04     NaN        
50%     5.898905e+07  0.000000     ...  4.404700e+04     NaN        
75%     5.899655e+07  0.000000     ...  1.005000e+05     NaN        
max     5.900624e+07  34.000000    ...  1.800005e+08     NaN        

[11 rows x 12 columns]

In [51]:
## Realizamos un pivot de la funcion describe para visualizar mejor nuestros datos!
df_historial.describe(include = 'all').T

count unique         top  ...         50%         75%          max
id               8250.0  NaN    NaN         ...  58989048.0  58996551.0  59006239.0 
OVD_t1           8250.0  NaN    NaN         ...  0.0         0.0         34.0       
OVD_t2           8250.0  NaN    NaN         ...  0.0         0.0         34.0       
OVD_t3           8250.0  NaN    NaN         ...  0.0         0.0         35.0       
OVD_sum          8250.0  NaN    NaN         ...  0.0         0.0         31500.0    
pay_normal       8250.0  NaN    NaN         ...  11.0        25.0        36.0       
prod_code        8250.0  NaN    NaN         ...  10.0        10.0        27.0       
prod_limit       2132.0  NaN    NaN         ...  68200.0     112200.0    660000.0   
update_date      8224    3041   30/10/2012  ...  NaN         NaN         NaN        
new_balance      8250.0  NaN    NaN         ...  0.0         24948.0     163211958.0
highest_balance  7841.0  NaN    NaN         ...  44047.0     100500.0    180000500.0
report_date      7136    1862   21/09/2015  ...  NaN         NaN         NaN        

[12 rows x 11 columns]

In [52]:
## Chequeamos la cantidad de ids que son unicos en el df.
df_historial['id'].nunique()
## Arrojando como resultado 1125 valores de 'id' únicos al igual que el set de clientes,
## lo cual a priori indicaria que ambos datasets tienen la mismos ids
## Es necesario realizar un paso extra para corroborar 
## ya que pueden coincidir en cantidad pero ser distintos.



1125

Arrojando como resultado 1125 valores de 'id' únicos al igual que el set de clientes,lo cual a priori indicaría que ambos datasets tienen la mismos 'ids'
Es necesario realizar un paso extra para corroborar ya que pueden coincidir en cantidad pero ser distintos.


In [53]:
## Buscamos si hay valores nulos
df_historial.isnull().values.any()


True

Al ser 'True' hay que tomar una decision respecto al tratamiento de los valores nulos.

In [ ]:
##Buscamos en que columna hay valores nulos

#df_historial['id'].isnull().values.any() # Arrojo como resultado False
#df_historial['OVD_t1'].isnull().values.any() # Arrojo como resultado False
#df_historial['OVD_t2'].isnull().values.any() # Arrojo como resultado False
#df_historial['OVD_t3'].isnull().values.any() # Arrojo como resultado False
#df_historial['OVD_sum'].isnull().values.any() # Arrojo como resultado False
#df_historial['pay_normal'].isnull().values.any() # Arrojo como resultado False
#df_historial['prod_code'].isnull().values.any() # Arrojo como resultado False
#df_historial['prod_limit'].isnull().values.any() # Arrojo como resultado True
#df_historial['update_date'].isnull().values.any() # Arrojo como resultado True
#df_historial['new_balance'].isnull().values.any() # Arrojo como resultado False
#df_historial['highest_balance'].isnull().values.any() # Arrojo como resultado True
#df_historial['report_date'].isnull().values.any() # Arrojo como resultado True




False

> El campo 'prod_limit' tiene 2132 valores mientras que los demas campos contienen, en general, 8250 valores. 

>> El campo 'prod_limit' corresponde a un valor flotante e indica el limite de un producto especifico, el problema reside en que este limite varia segun las caracteristicas de riesgo que el banco concluya sobre una persona, reemplazar un valor vacio en este campo seria, cuanto menos, riesgoso, es preferible truncar el set de datos a estos 2132 valores.

>>update_date = Fecha de actualizacion de la cuenta
highest_balance = Mediante excel se corrobora que aunque haya valores missings, no se han encontrado id's que no tengan alun ingreso en este campo, deberia reemplazarse por el valor mas alto segun ID. Se debera analizar los casos donde no se tiene un highest balance, si es que hay casos luego.

>>report_date = no debe ser reemplazada ya que es la ultima fecha de pago de la cuenta y puede no haber nunca un ultimo pago.
